### PR TITLE
#4703 Add size to rules

### PR DESCRIPTION
**Pull request by 3rdvision**
| - | - |
| --- | --- |
| Issue | https://github.com/ripe-tech/ripe-core/issues/4703 |
| Dependencies | https://github.com/ripe-tech/ripe-core/pull/4707 |
| Decisions | - Adapt ripe-sdk methods to new size dimensions in rules. <br> - Add missing resolveTransportRule and resolveTransportRuleP methods.  |

### CODE REVIEW

**Review by BeeMargarida — CHANGES_REQUESTED** [review]
Shouldn't this be done for `transport-rule` and `availability-rule` as well?

**Comment by 3rdvision** [comment]
> Shouldn't this be done for `transport-rule`, `availability-rule` and `price-rule` as well?

for which methods? Everything is passed through options and as for the resolve method, there's only `resolveInvoiceRule`

**Comment by BeeMargarida** [comment]
> > Shouldn't this be done for `transport-rule`, `availability-rule` and `price-rule` as well?
> 
> for which methods? Everything is passed through options and as for the resolve method, there's only `resolveInvoiceRule`

Nvm, i did not notice those don't have resolve methods. However, `transport-rule` should have (I guess it wasn't ported at the time) because there is an endpoint for it - https://github.com/ripe-tech/ripe-core/blob/7a0a2eed80467916980ecd02f487d307ca5cfa60/src/ripe_core/controllers/api/rules/transport_rule.py#L49
Perhaps this PR should add it, since it is basically the same as the one for invoice rule

File: src/js/api/transport-rule.js [hunk 1]
```diff
@@ -194,3 +194,87 @@ ripe.Ripe.prototype.deleteTransportRuleP = function(id, options) {
         });
     });
 };
+
+/**
+ * Gets an existing transport rule filtered by brand, model and country.
```
Comment (BeeMargarida) [inline src/js/api/transport-rule.js:199]:
```suggestion
 * Gets an existing transport rule filtered by brand, model, country, factory and size.

```

**Review by BeeMargarida — APPROVED** [review]

**Review by joao-conde — APPROVED** [review]

**Comment by 3rdvision** [comment]
blocked until https://github.com/ripe-tech/ripe-core/pull/4707 is merged

**Review by joamag — APPROVED** [review]

**Pull request merged**